Content.Shared/Emag/Systems/EmagSystem.cs

```diff
@@ -156,12 +156,28 @@ public bool TryUseEmag(EntityUid emag, EntityUid user, EntityUid target, EmagCom
         /// </summary>
         public bool DoEmagEffect(EntityUid user, EntityUid target)
         {
+            // prevent emagging twice
+            if (IsEmagged(target))
+                return false;
+
             var emaggedEvent = new GotEmaggedEvent(user);
             RaiseLocalEvent(target, ref emaggedEvent);
+
+            // add marker component unless the handler wants repeatable emagging
+            if (emaggedEvent.Emag)
+                EntityManager.AddComponent<EmaggedComponent>(target);
             return emaggedEvent.Handled;
         }
+
+        /// <summary>
+        /// Returns whether an entity has the emagged marker component
+        /// </summary>
+        public bool IsEmagged(EntityUid uid)
+        {
+            return EntityManager.HasComponent<EmaggedComponent>(uid);
+        }
```

Comment:
Why is this needed? You can just do HasComp directly or use EntityQuery to check for marker comps.

Reply:
just a little helper i thought i would use it more, can remove